src/engine/Service.cpp

```diff
@@ -122,60 +120,100 @@ Result Service::computeResult([[maybe_unused]] bool requestLaziness) {
             << ", target: " << serviceUrl.target() << ")" << std::endl
             << serviceQuery << std::endl;
 
-  // Send the query to the remote SPARQL endpoint via a POST request and get the
-  // result as TSV.
-  //
-  // TODO: We ask for the result as TSV because that is a compact and
-  // easy-to-parse format. It might not be the best choice regarding robustness
-  // and portability though. In particular, we are not sure how deterministic
-  // the TSV output is with respect to the precise encoding of literals.
-  cppcoro::generator<std::span<std::byte>> tsvByteResult =
-      ad_utility::reChunkAtSeparator(
-          getTsvFunction_(serviceUrl, cancellationHandle_,
-                          boost::beast::http::verb::post, serviceQuery,
-                          "application/sparql-query",
-                          "text/tab-separated-values"),
-          static_cast<std::byte>('\n'));
-
-  // TODO<GCC12> Use `std::views::transform` instead.
-  auto tsvResult = [](auto byteResult) -> cppcoro::generator<std::string_view> {
+  cppcoro::generator<std::span<std::byte>> jsonByteResult = getResultFunction_(
+      serviceUrl, cancellationHandle_, boost::beast::http::verb::post,
+      serviceQuery, "application/sparql-query",
+      "application/sparql-results+json");
+
+  auto interResult = [](auto byteResult) -> cppcoro::generator<std::string> {
     for (std::span<std::byte> bytes : byteResult) {
-      co_yield std::string_view{reinterpret_cast<const char*>(bytes.data()),
-                                bytes.size()};
+      co_yield std::string{reinterpret_cast<const char*>(bytes.data()),
+                           bytes.size()};
     }
-  }(std::move(tsvByteResult));
+  }(std::move(jsonByteResult));
+
+  std::string jsonStr;
+  for (const auto& s : interResult) {
+    absl::StrAppend(&jsonStr, s);
```

Comment:
Get rid of the intermediate interRes generator, and directly put the `reinterpret_cast<string_view....>` here.